syncd/RedisClient.cpp

```diff
@@ -741,6 +797,57 @@ sai_object_id_t RedisClient::getVidForRid(
     return vid;
 }
 
+void RedisClient::getVidForRid(
+        _In_ const sai_object_id_t* rids,
+        _Out_ sai_object_id_t* vids,
+        _In_ size_t count)
+{
+    SWSS_LOG_ENTER();
+
+    swss::RedisCommand hmget;
+
+    std::vector<std::string> cmds;
+    cmds.reserve(count + 2);
+
+    cmds.push_back("HMGET");
+    cmds.push_back(RIDTOVID);
+
+    for (size_t idx = 0; idx < count; idx++)
+    {
+        cmds.push_back(sai_serialize_object_id(rids[idx]));
+    }
+
+    hmget.format(cmds);
+
+    swss::RedisReply r(m_dbAsic.get(), hmget);
+
+    auto ctx = r.getContext();
+
+    if (ctx->type != REDIS_REPLY_ARRAY)
+    {
+        SWSS_LOG_THROW("expected array reply, got %d", ctx->type);
+    }
+
+    if (ctx->elements != count)
+    {
+        SWSS_LOG_THROW("expected %zu elements, got %zu", count, ctx->elements);
+    }
+
+    for (size_t idx = 0; idx < ctx->elements; idx++)
+    {
+        auto element = ctx->element[idx];
+
+        if (element->type == REDIS_REPLY_STRING)
+        {
+            sai_deserialize_object_id(element->str, vids[idx]);
+        }
+        else
```

Comment:
in single version of this function null is returned when hget returns nullptr, in this case what is REDIS_REPLY_XXX  value? maybe only one specific should be considered, and other should throw since redis can reply multple errors

Reply:
Done